From shimoda.yoshihiro@renesas.com  Fri Jun 27 14:35:05 2008
From: Yoshihiro Shimoda <shimoda.yoshihiro@renesas.com>
Date: Fri, 27 Jun 2008 19:09:55 +0900
Subject: usb: r8a66597-hcd: fix interrupt trigger
To: gregkh@suse.de
Cc: linux-usb@vger.kernel.org
Message-ID: <4864BC73.4050607@renesas.com>


fix the problem that did not set IRQF_TRIGGER_ flag.

Signed-off-by: Yoshihiro Shimoda <shimoda.yoshihiro@renesas.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/r8a66597-hcd.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/drivers/usb/host/r8a66597-hcd.c
+++ b/drivers/usb/host/r8a66597-hcd.c
@@ -2244,6 +2244,7 @@ static int __init r8a66597_probe(struct 
 	struct r8a66597 *r8a66597;
 	int ret = 0;
 	int i;
+	unsigned long irq_trigger;
 
 	if (pdev->dev.dma_mask) {
 		ret = -EINVAL;
@@ -2302,7 +2303,11 @@ static int __init r8a66597_probe(struct 
 	INIT_LIST_HEAD(&r8a66597->child_device);
 
 	hcd->rsrc_start = res->start;
-	ret = usb_add_hcd(hcd, irq, IRQF_DISABLED);
+	if (irq_sense == INTL)
+		irq_trigger = IRQF_TRIGGER_LOW;
+	else
+		irq_trigger = IRQF_TRIGGER_FALLING;
+	ret = usb_add_hcd(hcd, irq, IRQF_DISABLED | irq_trigger);
 	if (ret != 0) {
 		err("Failed to add hcd");
 		goto clean_up;
